src/shr_pio_mod.F90

```diff
@@ -53,225 +41,22 @@ module shr_pio_mod
      integer :: pio_netcdf_ioformat
   end type pio_comp_t
 
-  character(len=16), allocatable :: io_compname(:)
-  type(pio_comp_t), allocatable :: pio_comp_settings(:)
-  type (iosystem_desc_t), allocatable, target :: iosystems(:)
-  integer :: io_comm
-  logical :: pio_async_interface
-  integer, allocatable :: io_compid(:)
-  integer :: pio_debug_level=0, pio_blocksize=0
-  integer(kind=pio_offset_kind) :: pio_buffer_size_limit=-1
-  integer :: pio_rearr_opt_comm_type, pio_rearr_opt_fcd
-  logical :: pio_rearr_opt_c2i_enable_hs, pio_rearr_opt_c2i_enable_isend
-  integer :: pio_rearr_opt_c2i_max_pend_req
-  logical :: pio_rearr_opt_i2c_enable_hs, pio_rearr_opt_i2c_enable_isend
-  integer :: pio_rearr_opt_i2c_max_pend_req
-  integer :: total_comps=0
-
-#define DEBUGI 1
-
-#ifdef DEBUGI
-  integer :: drank
-#endif
+  !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  ! The following variables are public just for the sake of the driver's init_pio_mod.
+  ! No other subroutines should access these directly! Any other access should go
+  ! through the above accessor routines.
```

Comment:
Making the variables private to `shr_pio_mod` would have required adding a number of additional getter and setter routines. This seems doable but tedious and would add complexity.